### ISSUE TITLE
Objective-Sharpie not working anymore, needs .NET 6 update.

### ISSUE THREAD

**Issue by blazorin** [dotnet/maui#7755]
### Description

Using objective-sharpie bindings is not working anymore since it has been kept on the old SDK (Xamarin.iOS) instead of Microsoft.iOS and thus it gives compile errors related to NSObject and so.

Sharpie still needs to use the legacy SDK and it's a block to bind any objective-c library.

### Steps to Reproduce

Create a .NET Maui APP.
Create a iOS Binding project
 - Add a native reference to the library.
 - Add ApiDefinitions and ApiEnums.
 
From the Maui project add reference to the binding project.

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15 .NET 6

### Did you find any workaround?

The legacy SDK was at least on Release Candidate 1 (September 2021)

**Comment by drasticactions** [comment]
This isn't an issue that can be fixed in this repo, as it's not a MAUI (The UI Framework) issue. I'm not sure if that's a [Xamarin.MaciOS](https://github.com/xamarin/xamarin-macios) issue to be tracked, or where it should go.

@dalexsoto @rolfbjarne Do ya'll know where this should go?

I would think you can still use the generated bindings it makes and move them to a newer, net6 based, binding (`dotnet new iosbinding`). You would need to adjust the types, but it should work.

**Comment by dalexsoto** [comment]
Hello!

> ### Steps to Reproduce
> Create a .NET Maui APP. Add a native reference to the library. Add ApiDefinitions and ApiEnums.

This sounds wrong, you need to use a binding project using `dotnet new iosbinding` there you would add your ApiDefinitions and ApiEnums and reference the resulting dll in your main project.

**Comment by teranpeterson** [comment]
I am experiencing this same issue. Types are all conflicting because they are looking for the Xamarin.iOS.xxx instead of the Microsoft.iOS.xxx. Did this get reported upstream?

**Comment by teranpeterson** [comment]
To be more specific (and I believe this is the same for @blazorin), the `Xamarin.iOS` dependency required by `Xamarin Binding Library (iOS)` projects is the problem. 

Objective Sharpie still works to automatically create `ApiDefinitions.cs` and `StructsAndEnums.cs` files. These files can then be added to a binding library project and it all works quite well with Xamarin.iOS and Xamarin.Forms. 

The problem arises when a binding library is used in .NET Maui. All the bindings appear as usual but any iOS types referenced in the bound code throw errors. 

Error:
```
You must add a reference to assembly 'Xamarin.iOS, Version=0.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065'.
```

Examples:
```c#
/// Class generated in binding library
public class MyViewController : UIViewController ...

/// Failed cast
UIViewController myView = new MyViewController();
```

```c#
/// Method generated in binding library
public static void DoSomething(UIImage image) ...

/// Failed cast
Foo.DoSomething(image);
```

The type system can't convert the types from Xamarin.iOS to Microsoft.iOS, even though they are in essence identical. And so far, I haven't found an easy way to remove this dependency. 

Interestingly, the `Xamarin Binding Library (Android)`, does work flawlessly with .NET Maui. It depends on Mono.Android instead of Xamarin.Android.

**Comment by teranpeterson** [comment]
A solution was released yesterday as part of `.NET Conf: Focus on Maui` in [this](https://www.youtube.com/watch?v=oibfI-ZsmzQ) video. There are new binding library templates that can be built from the cli:

```bash
dotnet new iosbinding
```

The above command almost works, though I had to make several tweaks to get everything to build right. The .csproj is missing a few xml tags and VS for Mac doesn't support building it yet. I was able to successfully migrate a `Xamarin Binding Library (iOS)` project as follows:

1. Open existing Xamarin Binding Library project
2. Delete the `Properties/AssemblyInfo.cs` file
3. Replace the contents of the .csproj file with the following

```xml
<?xml version="1.0" encoding="utf-8"?>
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net6.0-ios</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>true</ImplicitUsings>
    <IsBindingProject>true</IsBindingProject>
    <!--This part isn't included in the new binding template yet -->
    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
    <NoBindingEmbedding>false</NoBindingEmbedding>
    <!-- End addition -->
  </PropertyGroup>
  <ItemGroup>
    <ObjcBindingApiDefinition Include="ApiDefinitions.cs" />
    <ObjcBindingCoreSource Include="StructsAndEnums.cs" />
  </ItemGroup>
  <ItemGroup>
    <NativeReference Include="/path/to/.framework">
      <Kind>Framework</Kind>
      <ForceLoad>True</ForceLoad>
      <SmartLink>False</SmartLink>
      <Frameworks>...</Frameworks>
    </NativeReference>
  </ItemGroup>
</Project>

```

4. From the cli, run `dotnet build /path/to/.csproj`
5. Use the .dll file produced as before

@kinfey has some better documentation on the process [here](https://github.com/kinfey/AMapMAUIControls). 

TLDR; the tooling is just barely being started. With some elbow grease it works for now. Give it time and VS for Mac will eventually have support for it.

**Comment by blazorin** [comment]
> A solution was released yesterday as part of `.NET Conf: Focus on Maui` in [this](https://www.youtube.com/watch?v=oibfI-ZsmzQ) video. There are new binding library templates that can be built from the cli:
> 
> ```shell
> dotnet new iosbinding
> ```
> 
> The above command almost works, though I had to make several tweaks to get everything to build right. The .csproj is missing a few xml tags and VS for Mac doesn't support building it yet. I was able to successfully migrate a `Xamarin Binding Library (iOS)` project as follows:
> 
> 1. Open existing Xamarin Binding Library project
> 2. Delete the `Properties/AssemblyInfo.cs` file
> 3. Replace the contents of the .csproj file with the following
> 
> ```
> <?xml version="1.0" encoding="utf-8"?>
> <Project Sdk="Microsoft.NET.Sdk">
>   <PropertyGroup>
>     <TargetFramework>net6.0-ios</TargetFramework>
>     <Nullable>enable</Nullable>
>     <ImplicitUsings>true</ImplicitUsings>
>     <IsBindingProject>true</IsBindingProject>
>     
>     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
>     <NoBindingEmbedding>false</NoBindingEmbedding>
>     
>   </PropertyGroup>
>   <ItemGroup>
>     <ObjcBindingApiDefinition Include="ApiDefinitions.cs" />
>     <ObjcBindingCoreSource Include="StructsAndEnums.cs" />
>   </ItemGroup>
>   <ItemGroup>
>     <NativeReference Include="/path/to/.framework">
>       <Kind>Framework</Kind>
>       <ForceLoad>True</ForceLoad>
>       <SmartLink>False</SmartLink>
>       <Frameworks>...</Frameworks>
>     </NativeReference>
>   </ItemGroup>
> </Project>
> ```
> 
> 4. From the cli, run `dotnet build /path/to/.csproj`
> 5. Use the .dll file produced as before
> 
> @kinfey has some better documentation on the process [here](https://github.com/kinfey/AMapMAUIControls).
> 
> TLDR; the tooling is just barely being started. With some elbow grease it works for now. Give it time and VS for Mac will eventually have support for it.

Good to see this has been solved.
Altough I needed to move forward and have not used .NET to keep building the app.

**Comment by KSemenenko** [comment]
I found the same issue

**Comment by KSemenenko** [comment]
```
  The imported project "/usr/local/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/Sdk/Sdk.props" was not found. Confirm that the expression in the Import declaration "/usr/local/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/Sdk/Sdk.props" is correct, and that the file exists on disk.  /Users/Developer/AgoraSDK/AgoraIOS/AgoraIOS.csproj
```

```
  <PropertyGroup>
    <TargetFramework>net7.0-ios</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>true</ImplicitUsings>
    <IsBindingProject>true</IsBindingProject>
    <SupportedOSPlatformVersion>13.0</SupportedOSPlatformVersion>
  </PropertyGroup>
```

why net6.0?

**Comment by mattleibow** [comment]
This issue was moved to xamarin/xamarin-macios#17449
